File: 2125/cpp/solution.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int res = 0;
        int prev = 0;

        for (string row: bank){
            int count = 0;
            for (char c: row){
                if (c == '1'){
                    count++;
                }
            }

            res += count * prev;
            if (count > 0){
                prev = count;
            }
            
        }
        return res;
    }
};

int main(){
    Solution sol = Solution();
    vector<string> test = {"0"};
    sol.numberOfBeams(test);
}

File: 2125/rust/solution.rs
struct Solution {}

impl Solution {
    pub fn number_of_beams(bank: Vec<String>) -> i32 {
        let mut rows: Vec<i32> = Vec::new();

        let mut prev = 0;
        let mut res = 0;

        for row in bank{
            let mut count = 0;
            for c in row.chars(){
                if c == '1'{
                    count += 1;
                }
            }
            res += prev *count;
            if count > 0{
                prev = count;
            }
        }
        return res;
    }
}

fn main(){
    let res = Solution::number_of_beams(vec!["0".to_string()]);

    println!("Solution: {res}");
}